flake.nix

```diff
@@ -5,99 +5,131 @@
+                      haskellPackages = hf;
+                      extraLibraries = [ llvmPackages.llvm.dev ];
+                      overrideAttrs = o: {
+                        checkPhase = ''
                           runHook preCheck
-                          DATALOG_DIR="${o.src}/cbits/" SOUFFLE_BIN="${pkgs.souffle}/bin/souffle" ./Setup test
+                          DATALOG_DIR="${self}/cbits/" SOUFFLE_BIN="${pkgs.souffle}/bin/souffle" ./Setup test
                           runHook postCheck
-                          '';
-                        });
+                        '';
+                      };
                     };
-                };
-            in { inherit haskellPackages; };
-          pkgs = import np {
-            inherit system config;
-            overlays = [
-              ds.overlay
-              shs.overlay."${system}"
-              llvm-cg.overlay."${system}"
-              overlay
-            ];
-          };
-        in with pkgs.lib; rec {
-          inherit overlay;
-          packages = { inherit (pkgs.haskellPackages) eclair-lang; };
-          defaultPackage = packages.eclair-lang;
-          devShell = pkgs.devshell.mkShell {
+                  };
+              };
+          in { inherit haskellPackages llvmPackages; };
+        pkgs = import np {
+          inherit system config;
+          overlays = [ ds.overlay shs.overlay."${system}" overlays.default ];
+        };
+        packages = rec {
+          inherit (pkgs.haskellPackages) eclair-lang;
+          default = eclair-lang;
+        };
+        devShells = rec {
+          default = pkgs.devshell.mkShell {
             name = "ECLAIR-LANG";
             imports = [ (pkgs.devshell.importTOML ./devshell.toml) ];
             packages = with pkgs;
               with haskellPackages; [
                 souffle
-                pkgs.llvmPackages_14.llvm.dev
                 pkgs.ghcid
+                llvmPackages.libllvm
+                llvmPackages.llvm.dev
+                llvmPackages.bintools-unwrapped
                 (ghcWithPackages (p:
                   with p; [
                     algebraic-graphs
-                    hspec-discover
-                    llvm-codegen
-                    souffle-haskell
-                    ghc
                     cabal-install
+                    ghc
                     hsc2hs
                     hpack
                     haskell-language-server
                     hlint
+                    hspec-discover
+                    llvm-codegen
+                    souffle-haskell
                   ]))
               ];
             # Next line always sets DATALOG_DIR so souffle can find the datalog files in interpreted mode.
-            env = [{ name = "DATALOG_DIR"; value = "cbits/"; }];
+            env = [{
+              name = "DATALOG_DIR";
+              value = "cbits/";
+            }];
           };
-        });
+        };
+      in { inherit overlays packages devShells; });
```

Comment:
@smunix I think you made a mistake here? Because some keys have a ".default" key in between..

Reply:
No mistake, you need to update your local nix version. See the changes I made earlier in the GitHub action `build.yaml` file?
The current stake version of Nix is `2.11.0`, yours is certainly way behind. Can you check?